### PR TITLE
Paril/filterimage

### PR DESCRIPTION
**Pull request by Paril**
Moves the FilterImage procedure for renderer images to the GPU.
Also some minor cleanup on the Windows side of things, like including OpenAL DLL in the package.

### CODE REVIEW

File: src/client/renderer/r_bsp_model.c [hunk 1]
```diff
@@ -850,9 +850,9 @@ void R_ExportBSP_f(void) {
 			}
 
 			num_surfs++;
-
+	
 			const uint32_t *element = surf->elements + surf->num_edges - 1;
-
+
```
Comment (jdolan) [inline src/client/renderer/r_bsp_model.c]: Bad whitespace diffs

File: src/client/renderer/r_image.c [hunk 2]
```diff
@@ -146,43 +146,19 @@ void R_Screenshot_f(void) {
...
-		if (r_monochrome->integer & mask) { // monochrome
-			p[0] = p[1] = p[2] = (p[0] + p[1] + p[2]) / 3;
-		}
-
-		if (r_invert->integer & mask) { // inverted
-			p[0] = 255 - p[0];
-			p[1] = 255 - p[1];
-			p[2] = 255 - p[2];
-		}
-
```
Comment (jdolan) [inline src/client/renderer/r_image.c]: 👍

File: src/client/renderer/shaders/default_fs.glsl [hunk 3]
```diff
@@ -10,9 +10,10 @@
 #include "include/fog.glsl"
 #include "include/noise3d.glsl"
 #include "include/tint.glsl"
+#include "include/color.glsl"
 
 #define MAX_LIGHTS $r_max_lights
-#define LIGHT_SCALE $r_modulate
+#define MODULATE_SCALE $r_modulate
```
Comment (jdolan) [inline src/client/renderer/shaders/default_fs.glsl]: Yea, or just `MODULATE`

File: src/client/renderer/shaders/default_fs.glsl [hunk 4]
```diff
@@ -250,7 +251,7 @@ void main(void) {
 	vec4 diffuse = vec4(1.0);
 
 	if (DIFFUSE) { // sample the diffuse texture, honoring the parallax offset
-		diffuse = texture(SAMPLER0, texcoords[0] + parallax);
+		diffuse = ColorFilter(texture(SAMPLER0, texcoords[0] + parallax));
```
Comment (jdolan) [inline src/client/renderer/shaders/default_fs.glsl]: Awesome. So clean.

File: src/client/renderer/shaders/include/color.glsl [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+#endif
+
+#if INVERT
+	color.rgb = vec3(1.0) - color.rgb;
+#endif
+
+	color.rgb = mix(vec3(0.5), mix(vec3(dot(vec3(0.2125, 0.7154, 0.0721), color.rgb * BRIGHTNESS_SCALE)), color.rgb * BRIGHTNESS_SCALE, SATURATION_SCALE), CONTRAST_SCALE);
+
+	return color;
+}
```
Comment (jdolan) [inline src/client/renderer/shaders/include/color.glsl]: Tight. I think most of our shaders have some comments, but I won't hold up a merge over it.

**Comment by jdolan** [comment]
Any reason to not merge this? Or should we wait to see what Spiney's changes look like first?

**Comment by jdolan** [comment]
*nudge*

**Comment by jdolan** [comment]
What's the deal with this branch? Is this no longer needed? Or is there any reason to not merge it?

**Comment by Paril** [comment]
This can be merged I think. Just have to resolve the lil conflicts and it's OK to go.

**Comment by jdolan** [comment]
If I could resolve those conflicts, I would :P

**Pull request merged**